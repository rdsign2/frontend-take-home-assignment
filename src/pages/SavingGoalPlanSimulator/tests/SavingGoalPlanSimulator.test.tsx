import * as React from 'react';
import { shallow } from 'enzyme';
import SGPS from '../index';

// import {
//   // months,
//   // currentYear,
//   // currentMonth,
//   // formatter,
//   // minMonthGoalPeriod,
//   // minStartingMonthIndex,
//   // minStartingMonth,
//   // disabledMonths,
//   changeDate
//   // pickDate
// } from '../date';

// const mockedData = {
//   totalAmount: 30000,
//   reachGoalBy: 11,
//   expectedMonthlyAmount: 10000
// };

describe('Saving goal plan simulator', () => {
  const component = shallow(<SGPS />);
  // console.log(component.debug());

  it('renders', () => {
    expect(component.exists()).toEqual(true);
  });

  it('shows correct title', () => {
    expect(component.find('h3').text()).toBe(`Let's plan your saving goal.`);
  });

  it('snapshotShot', () => {
    expect(component.getElements()).toMatchSnapshot();
  });
});

// const mockedData = {
//   totalAmount: 30000,
//   reachGoalBy: 11,
//   expectedMonthlyAmount: 10000
// };

// describe('Monthly amount value', () => {
//   it('the division between totalAmount by reachGoalBy should result in the expectedMonthlyAmount', () => {
//     const zz = changeDate()
//   })
// })
