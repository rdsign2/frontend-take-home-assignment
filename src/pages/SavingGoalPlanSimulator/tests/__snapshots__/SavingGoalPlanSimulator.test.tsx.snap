// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saving goal plan simulator snapshotShot 1`] = `
Array [
  <section
    className="SavingGoalPlanSimulator"
  >
    <h3
      className="tp"
    >
      Let's plan your 
      <strong>
        saving goal.
      </strong>
    </h3>
    <div
      className="goalPlanCalculator"
    >
      <div
        className="heading"
      >
        <IconHouse
          color="#657786"
        />
        <h1>
          Buy a house
        </h1>
        <p>
          Saving goal
        </p>
      </div>
      <div
        className="inputs row"
      >
        <Input
          type="number"
          valueSetter={[Function]}
        />
        <DatePicker
          currentMonth="September"
          currentYear={2020}
          disableds={
            Array [
              "January",
              "February",
              "March",
              "April",
              "May",
              "June",
              "July",
              "August",
              "September",
            ]
          }
          minStart="October"
          months={
            Array [
              "January",
              "February",
              "March",
              "April",
              "May",
              "June",
              "July",
              "August",
              "September",
              "October",
              "November",
              "December",
            ]
          }
          nextMonth={[Function]}
          nextYear={[Function]}
          pickedDate={[Function]}
          previousMonth={[Function]}
          previousYear={[Function]}
          stateMonth="October"
          stateYear={2020}
          validateYear={[Function]}
        />
      </div>
      <div
        className="summary"
      >
        <div
          className="row"
        >
          <p
            className="label"
          >
            Monthly
            <span>
               amount
            </span>
          </p>
          <h2
            className="amount tp"
          >
            $
            <strong>
              0
            </strong>
          </h2>
        </div>
        <div
          className="details row"
        >
          <p>
            Insert total amount and choose a date.
          </p>
        </div>
      </div>
      <Button
        enable={false}
      />
    </div>
  </section>,
]
`;
